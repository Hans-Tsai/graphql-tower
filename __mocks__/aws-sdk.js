export const promise = jest.fn(() => Promise.resolve());
export const pipe = jest.fn();

const createReadStream = params => ({ pipe: () => pipe(params) });

export default {
  S3: () => ({
    headObject: params => ({
      promise: () => promise({ ...params, method: 'headObject' }),
      createReadStream: () => createReadStream({ ...params, method: 'headObject' }),
    }),
    getObject: params => ({
      promise: () => promise({ ...params, method: 'getObject' }),
      createReadStream: () => createReadStream({ ...params, method: 'getObject' }),
    }),
    copyObject: params => ({
      promise: () => promise({ ...params, method: 'copyObject' }),
      createReadStream: () => createReadStream({ ...params, method: 'copyObject' }),
    }),
    upload: params => ({
      promise: () => promise({ ...params, method: 'upload' }),
      createReadStream: () => createReadStream({ ...params, method: 'upload' }),
    }),
  }),
};
