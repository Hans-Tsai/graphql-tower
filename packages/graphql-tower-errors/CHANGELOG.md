# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

<a name="3.21.3"></a>
## [3.21.3](https://github.com/tmotx/graphql-tower/compare/v3.21.2...v3.21.3) (2017-12-05)




**Note:** Version bump only for package graphql-tower-errors

<a name="3.16.0"></a>
# [3.16.0](https://github.com/tmotx/graphql-tower/compare/v3.15.1...v3.16.0) (2017-11-19)


### Features

* **help:** added assertResult ([e6cd019](https://github.com/tmotx/graphql-tower/commit/e6cd019))




<a name="3.15.1"></a>
## [3.15.1](https://github.com/tmotx/graphql-tower/compare/v3.15.0...v3.15.1) (2017-11-18)


### Bug Fixes

* **error:** added NotAcceptableError ([7258d43](https://github.com/tmotx/graphql-tower/commit/7258d43))
