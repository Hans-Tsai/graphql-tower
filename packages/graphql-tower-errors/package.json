{
  "name": "graphql-tower-errors",
  "version": "1.0.0",
  "description": "",
  "main": "lib/index.js",
  "scripts": {
    "build": "rm -rf lib/* && babel src --ignore __tests__ --out-dir lib"
  },
  "repository": {
    "type": "git",
    "url": "git+https://github.com/tmotx/graphql-tower.git"
  },
  "author": "tmotx",
  "license": "MIT",
  "bugs": {
    "url": "https://github.com/tmotx/graphql-tower/issues"
  },
  "homepage": "https://github.com/tmotx/graphql-tower#readme",
  "files": [
    "lib"
  ],
  "peerDependencies": {
    "graphql": "^0.11.7"
  },
  "devDependencies": {
    "faker": "^4.1.0"
  }
}
