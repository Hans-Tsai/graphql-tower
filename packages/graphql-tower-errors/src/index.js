import { GraphQLError } from 'graphql/error';

export class ForbiddenError extends GraphQLError {
  symbol = 'ForbiddenError';

  constructor(message = 'ForbiddenError') {
    super(message);
  }
}

export class NotFoundError extends GraphQLError {
  symbol = 'NotFoundError';

  constructor(message = 'NotFoundError') {
    super(message);
  }
}

export class UnauthorizedError extends GraphQLError {
  symbol = 'UnauthorizedError';

  constructor(message = 'UnauthorizedError') {
    super(message);
  }
}

export class PaymentRequiredError extends GraphQLError {
  symbol = 'PaymentRequiredError';

  constructor(message = 'PaymentRequiredError') {
    super(message);
  }
}

export class NotAcceptableError extends GraphQLError {
  symbol = 'NotAcceptableError';

  constructor(message = 'NotAcceptableError') {
    super(message);
  }
}

export class GoneDataError extends GraphQLError {
  symbol = 'GoneDataError';

  constructor(message = 'GoneDataError') {
    super(message);
  }
}

export class ConflictError extends GraphQLError {
  symbol = 'ConflictError';

  constructor(message = 'ConflictError') {
    super(message);
  }
}

export class LockedError extends GraphQLError {
  symbol = 'LockedError';

  constructor(message = 'LockedError') {
    super(message);
  }
}

export class UnavailableForLegalReasonsError extends GraphQLError {
  symbol = 'UnavailableForLegalReasonsError';

  constructor(message = 'UnavailableForLegalReasonsError') {
    super(message);
  }
}
