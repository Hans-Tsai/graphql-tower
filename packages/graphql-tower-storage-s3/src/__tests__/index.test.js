import { promise, pipe } from 'aws-sdk';
import { NotFoundError } from 'graphql-tower-errors';
import StorageS3 from '../';

jest.mock('graphql-tower-unique', () => () => 'UNIQUE_GUID');

describe('storage s3', () => {
  const storage = new StorageS3({ STORAGE_URL: 's3://username:secret_key@ap-northeast-1/graphql-tower' });

  it('is required', () => {
    expect(() => new StorageS3()).toThrowError(new TypeError('STORAGE_URL is required'));
    expect(() => new StorageS3({ STORAGE_URL: 's3://ap-northeast-1/graphql-tower' })).not.toThrowError();
  });

  describe('checkContentType', () => {
    it('when type is jpeg', async () => {
      promise.mockReturnValueOnce(Promise.resolve({ ContentType: 'image/jpeg' }));
      await storage.checkContentType('XYZ');
      expect(promise.mock.calls).toMatchSnapshot();
      expect(promise).toHaveBeenCalledTimes(1);
    });

    it('when type is png', async () => {
      promise.mockReturnValueOnce(Promise.resolve({ ContentType: 'image/png' }));
      await storage.checkContentType('XYZ');
      expect(promise.mock.calls).toMatchSnapshot();
      expect(promise).toHaveBeenCalledTimes(1);
    });

    it('when unsupported file type', async () => {
      promise.mockReturnValueOnce(Promise.resolve({ ContentType: 'application/epub+zip' }));
      await expect(storage.checkContentType('from XYZ')).rejects.toEqual(new TypeError('unsupported file type'));
      expect(promise).toHaveBeenCalledTimes(1);
    });

    it('when not found', async () => {
      promise.mockReturnValueOnce(Promise.reject(new Error('not found')));
      await expect(storage.checkContentType('XYZ')).rejects.toEqual(new TypeError('unsupported file type'));
      expect(promise).toHaveBeenCalledTimes(1);
    });
  });

  describe('upload', () => {
    it('successfully uploaded', async () => {
      pipe.mockReturnValueOnce('IMAGE_STREAM_BODY');
      await storage.upload('XYZ', 'IMAGE_UPLOAD_UUID');
      expect(promise.mock.calls).toMatchSnapshot();
      expect(pipe.mock.calls).toMatchSnapshot();
      expect(promise).toHaveBeenCalledTimes(2);
      expect(pipe).toHaveBeenCalledTimes(1);
    });

    it('successfully uploaded when use unique', async () => {
      pipe.mockReturnValueOnce('IMAGE_STREAM_BODY');
      await storage.upload('XYZ');
      expect(promise.mock.calls).toMatchSnapshot();
      expect(pipe.mock.calls).toMatchSnapshot();
      expect(promise).toHaveBeenCalledTimes(2);
      expect(pipe).toHaveBeenCalledTimes(1);
    });

    it('when not found', async () => {
      promise.mockReturnValueOnce(Promise.reject(new Error('not found')));
      await expect(storage.upload('XYZ')).rejects.toEqual(new NotFoundError());
      expect(promise).toHaveBeenCalledTimes(1);
      expect(pipe).toHaveBeenCalledTimes(0);
    });
  });

  describe('fetchCover', () => {
    it('successfully fetch', async () => {
      promise.mockReturnValueOnce(Promise.reject(new Error('not found')));
      pipe.mockReturnValueOnce('IMAGE_STREAM_BODY');

      const stream = await storage.fetchCover('IMAGE_KEY');
      expect(stream).toEqual({ pipe: expect.any(Function) });

      expect(promise.mock.calls).toMatchSnapshot();
      expect(pipe.mock.calls).toMatchSnapshot();
      expect(promise).toHaveBeenCalledTimes(2);
      expect(pipe).toHaveBeenCalledTimes(1);
    });

    it('successfully fetch use cache', async () => {
      const stream = await storage.fetchCover('IMAGE_KEY');
      expect(stream).toEqual({ pipe: expect.any(Function) });

      expect(promise.mock.calls).toMatchSnapshot();
      expect(promise).toHaveBeenCalledTimes(1);
      expect(pipe).toHaveBeenCalledTimes(0);
    });

    it('when input width and height', async () => {
      const stream = await storage.fetchCover('IMAGE_KEY', '2048', '2048');
      expect(stream).toEqual({ pipe: expect.any(Function) });
      expect(promise.mock.calls).toMatchSnapshot();
      expect(promise).toHaveBeenCalledTimes(1);
    });
  });

  describe('fetchPreCover', () => {
    it('successfully fetch', async () => {
      promise.mockReturnValueOnce(Promise.reject(new Error('not found')));
      pipe.mockReturnValueOnce('IMAGE_STREAM_BODY');

      const stream = await storage.fetchPreCover('IMAGE_KEY');
      expect(stream).toEqual({ pipe: expect.any(Function) });

      expect(promise.mock.calls).toMatchSnapshot();
      expect(pipe.mock.calls).toMatchSnapshot();
      expect(promise).toHaveBeenCalledTimes(2);
      expect(pipe).toHaveBeenCalledTimes(1);
    });

    it('successfully fetch use cache', async () => {
      const stream = await storage.fetchPreCover('IMAGE_KEY');
      expect(stream).toEqual({ pipe: expect.any(Function) });

      expect(promise.mock.calls).toMatchSnapshot();
      expect(promise).toHaveBeenCalledTimes(1);
      expect(pipe).toHaveBeenCalledTimes(0);
    });

    it('when input width and height', async () => {
      const stream = await storage.fetchPreCover('IMAGE_KEY', '2048', '2048');
      expect(stream).toEqual({ pipe: expect.any(Function) });
      expect(promise.mock.calls).toMatchSnapshot();
      expect(promise).toHaveBeenCalledTimes(1);
    });
  });
});
