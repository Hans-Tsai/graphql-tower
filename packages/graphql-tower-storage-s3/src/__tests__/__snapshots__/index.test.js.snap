// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`storage s3 checkContentType when type is jpeg 1`] = `
Array [
  Array [
    Object {
      "Key": "uploader/XYZ",
      "method": "headObject",
    },
  ],
]
`;

exports[`storage s3 checkContentType when type is png 1`] = `
Array [
  Array [
    Object {
      "Key": "uploader/XYZ",
      "method": "headObject",
    },
  ],
]
`;

exports[`storage s3 confirm successfully confirmed 1`] = `
Array [
  Array [
    Object {
      "CopySource": "graphql-tower/uploader/XYZ",
      "Key": "media/IMAGE_UPLOAD_UUID",
      "method": "copyObject",
    },
  ],
  Array [
    Object {
      "Body": "IMAGE_STREAM_BODY",
      "Key": "media/IMAGE_UPLOAD_UUID_cover",
      "method": "upload",
    },
  ],
]
`;

exports[`storage s3 confirm successfully confirmed 2`] = `
Array [
  Array [
    Object {
      "Key": "media/IMAGE_UPLOAD_UUID",
      "method": "getObject",
    },
  ],
]
`;

exports[`storage s3 confirm successfully confirmed when use unique 1`] = `
Array [
  Array [
    Object {
      "CopySource": "graphql-tower/uploader/XYZ",
      "Key": "media/UNIQUE_GUID",
      "method": "copyObject",
    },
  ],
  Array [
    Object {
      "Body": "IMAGE_STREAM_BODY",
      "Key": "media/UNIQUE_GUID_cover",
      "method": "upload",
    },
  ],
]
`;

exports[`storage s3 confirm successfully confirmed when use unique 2`] = `
Array [
  Array [
    Object {
      "Key": "media/UNIQUE_GUID",
      "method": "getObject",
    },
  ],
]
`;

exports[`storage s3 fetchCover successfully fetch 1`] = `
Array [
  Array [
    Object {
      "Key": "cache/IMAGE_KEY_cover_1920x",
      "method": "headObject",
    },
  ],
  Array [
    Object {
      "Body": "IMAGE_STREAM_BODY",
      "Key": "cache/IMAGE_KEY_cover_1920x",
      "method": "upload",
    },
  ],
]
`;

exports[`storage s3 fetchCover successfully fetch 2`] = `
Array [
  Array [
    Object {
      "Key": "media/IMAGE_KEY_cover",
      "method": "getObject",
    },
  ],
]
`;

exports[`storage s3 fetchCover successfully fetch use cache 1`] = `
Array [
  Array [
    Object {
      "Key": "cache/IMAGE_KEY_cover_1920x",
      "method": "headObject",
    },
  ],
]
`;

exports[`storage s3 fetchCover when input width and height 1`] = `
Array [
  Array [
    Object {
      "Key": "cache/IMAGE_KEY_cover_2048x2048",
      "method": "headObject",
    },
  ],
]
`;

exports[`storage s3 fetchPreCover successfully fetch 1`] = `
Array [
  Array [
    Object {
      "Key": "cache/IMAGE_KEY_precover_128x",
      "method": "headObject",
    },
  ],
  Array [
    Object {
      "Body": "IMAGE_STREAM_BODY",
      "Key": "cache/IMAGE_KEY_precover_128x",
      "method": "upload",
    },
  ],
]
`;

exports[`storage s3 fetchPreCover successfully fetch 2`] = `
Array [
  Array [
    Object {
      "Key": "media/IMAGE_KEY_cover",
      "method": "getObject",
    },
  ],
]
`;

exports[`storage s3 fetchPreCover successfully fetch use cache 1`] = `
Array [
  Array [
    Object {
      "Key": "cache/IMAGE_KEY_precover_128x",
      "method": "headObject",
    },
  ],
]
`;

exports[`storage s3 fetchPreCover when input width and height 1`] = `
Array [
  Array [
    Object {
      "Key": "cache/IMAGE_KEY_precover_128x2048",
      "method": "headObject",
    },
  ],
]
`;
